runtime/queries/c/highlights.scm

```diff
@@ -1,65 +1,102 @@
-(storage_class_specifier) @keyword.storage
-
-"goto" @keyword
-"register" @keyword
-"break" @keyword
-"case" @keyword
-"continue" @keyword
-"default" @keyword
-"do" @keyword
-"else" @keyword
-"enum" @keyword
-"extern" @keyword
-"for" @keyword
-"if" @keyword
-"inline" @keyword
-"return" @keyword
-"sizeof" @keyword
-"struct" @keyword
-"switch" @keyword
-"typedef" @keyword
-"union" @keyword
-"volatile" @keyword
-"while" @keyword
-"const" @keyword
+[
+  "sizeof"
+] @keyword
+
+[
+  "enum"
+  "struct"
+  "union"
+] @keyword.storage.type
+
+[
+  "const"
+  "extern"
+  "inline"
+  "register"
+  "typedef"
+  "volatile"
+  (storage_class_specifier)
+] @keyword.storage.modifier
+
+[
+  "for"
+  "do"
+  "while"
+  "break"
+  "continue"
+] @keyword.control.repeat
 
 [
- "#define"
- "#elif"
- "#else"
- "#endif"
- "#if"
- "#ifdef"
- "#ifndef"
- "#include"
- (preproc_directive)
+  "goto"
+  "if"
+  "else"
+  "switch"
+  "case"
+  "default"
```

Comment:
this will result in slight mishighlighting in C++:
`struct X { X() = default; };`

You could override it with 
```
(default_method_clause "default" @keyword) ; not @keyword.control
```